Add MiniTalk

vbrabandt opened the pull request:
PR request for MiniTalk

vbrabandt committed:
Create README.md

vbrabandt committed:
made a makefile and stuff

vbrabandt committed:
Merge branch 'main' into Add-MiniTalk

vbrabandt committed:
Added client.c, server.c and Libft to MiniTalk

vbrabandt committed:
Update Makefile

vbrabandt committed:
Merge branch 'main' into Add-MiniTalk

vbrabandt committed:
Updated Makefile

vbrabandt committed:
Update Makefile

Kinda finished it, well I need to do the .c stuff now :cry:

vbrabandt committed:
Probably completed MiniTalk.h

vbrabandt committed:
Added Capybaras

vbrabandt committed:
Updated MiniTalk.h and added it to .C files

vbrabandt committed:
Makefile now works as expected, but....

For some reason I have to add the path to MiniTalk.h directly, Idk I found this behaviour to be weird

vbrabandt committed:
Update Makefile

vbrabandt committed:
Update Makefile

vbrabandt committed:
Update Makefile

vbrabandt committed:
First try

vbrabandt committed:
Makefile update (again), and not working srcs

vbrabandt committed:
Still not working

:(

vbrabandt committed:
me cry in fixes tries

vbrabandt committed:
Update README.md

vbrabandt committed:
Half working bonus

vbrabandt committed:
Update Makefile

vbrabandt committed:
baba

vbrabandt committed:
Kinda working normal source version

vbrabandt2005 commented:
# GitHub Copilot Summery

### Author Information Updates:
* Updated author email address from `vbrabandt@student.42.fr` to `vbrabandt@proton.me` in various files:
  * `get_next_line.c`
  * `get_next_line.h`
  * `get_next_line_bonus.c`
  * `get_next_line_bonus.h`
  * `get_next_line_utils.c`
  * `get_next_line_utils_bonus.c`
  * `main.c`
  * `main_bonus.c`
  * `ft_printf-char.c`
  * `ft_printf-num.c`
  * `ft_printf.c`
  * `ft_printf.h`
  * `ft_printf-2024/Makefile`
  * `ft_printf-2024/main.c`

### MiniTalk Project Additions:
* Added `Makefile` for MiniTalk project, including compilation rules for client and server executables, along with their bonus versions.
* Created `README.md` for MiniTalk project with a brief description.
* Added `MiniTalk.h` header file with necessary includes and a macro definition.

### Libft_Modified Project Additions:
* Added `Makefile` for the `Libft_Modified` library, detailing the compilation process for various utility functions.
* Created `README.md` for `Libft_Modified` with a brief description of the library and its additional functions. I have got a kinda messy version working, but I found a de facto limit to the Linux System here with about 4000 words, I found 2000 works best for testing

vbrabandt2005 commented:
**Update:**
I have found a flaw with this project overall, not necessarily with the code, maybe it worked on the proper UNIX MacOS before, but since we migrated to Linux, MiniTalk now has a de-facto tested limit of 4000 words (from the Lorem ipsum).

I have seen other peoples (including one from a staff member) MiniTalk which had passed evaluation and all of them seem to have the same issue.

vbrabandt committed:
New working version

I got inspired by a random version of this on GitLab (Will credit later), all version of MiniTalk I have seen has a de-facto length limit, I will try to explain to my evaluators on this, but ye I'm hoping to finish this by new week

vbrabandt committed:
Merge branch 'main' into Add-MiniTalk

vbrabandt committed:
README updates and fixed Atoi

- Wrote a cool README.md
- Replaced my old `ft_atoi.c` specifically for MiniTalk
- Fixed the old `ft_printf` things idk
- Made a screenshot of a successful MiniTalk run (2000 words Lorum Ipsum)

Bonus:
I tabbed out some parts of the So_Long and MiniTalk READMEs but It probably doesn't matter much

vbrabandt committed:
Updated README and new pic

- Added turtle
- wrote proper stuff idk

vbrabandt committed:
Update README.md

vbrabandt committed:
Update README.md

vbrabandt2005 commented:
> **Update:** I have found a flaw with this project overall, not necessarily with the code, maybe it worked on the proper UNIX MacOS before, but since we migrated to Linux, MiniTalk now has a de-facto tested limit of 4000 words (from the Lorem ipsum).
> 
> I have seen other peoples (including one from a staff member) MiniTalk which had passed evaluation and all of them seem to have the same issue.

Ok I find out something weird, it also depends on the situation idk, sometimes 3000 words work but sometime it doesn't why does life have to be so hard :(

vbrabandt committed:
minor improvements

- Cute PID server announcement
- Removed global variable from server.c
- Norminette fixes in header file
- Silenced the opposition that is the compiling of normal Client and Server in the Makefile

vbrabandt committed:
just copied over normal to bonus (WIP)

FOR BONUS!!!!

vbrabandt committed:
Gave up on bonus tbh

vbrabandt2005 commented:
I decided to end my progress here, I just found this project is not really worth my time, so Bonus (Specifically Server message received responds is incomplete), but other than that everything mandatory should be fine.

Concerning my complaints of the Code not working properly (likely because of Linux's implementation of the Signals is different to Unix), I have found occasions where bigger amounts of characters work, I presume it's just being effected by the other system process.

vbrabandt committed:
Update README.md

vbrabandt committed:
Update README.md
